### PR TITLE
Upgrade ktor-swagger-ui

### PR DESCRIPTION
**Pull request by mnonnenmacher**
Upgrade ktor-swagger-ui to v3. As the new version now correctly marks properties as nullable, also do some necessary API refactorings. Please see the commit messages for details.

### CODE REVIEW

**Comment by mnonnenmacher** [comment]
@Etsija Please have a look at the last commit which contains some small changes to the UI code.

File: api/v1/model/src/commonMain/kotlin/OptionalValue.kt [hunk 1]
```diff
@@ -97,6 +90,15 @@ sealed interface OptionalValue<out T> {
         }
 }
 
+/**
+ * Return the [value][Present.value] if this [OptionalValue] is [Present], otherwise throw an
+ * [IllegalArgumentException].
+ */
+val <T> OptionalValue<T>.valueOrThrow: T get() {
```
Comment (sschuberth) [inline api/v1/model/src/commonMain/kotlin/OptionalValue.kt:97]: Would the [`@Transient` annotation](https://github.com/Kotlin/kotlinx.serialization/blob/master/docs/basic-serialization.md#transient-properties) maybe work as well?

File: model/src/commonMain/kotlin/JobConfigurations.kt [hunk 2]
```diff
@@ -80,7 +80,7 @@ data class AnalyzerJobConfiguration(
     /**
      * The list of package curation providers to use.
      */
-    val packageCurationProviders: List<ProviderPluginConfiguration> = emptyList(),
+    val packageCurationProviders: List<ProviderPluginConfiguration>? = null,
```
Comment (oheger-bosch) [inline model/src/commonMain/kotlin/JobConfigurations.kt:83]: I find it strange that we need to add quirks to our data model, just to make the Swagger generation work. If this is really required, should this change then be limited to the API module? There is no need for the property in `model` to be nullable, too.

File: model/src/commonMain/kotlin/JobConfigurations.kt [hunk 3]
```diff
@@ -191,7 +191,7 @@ data class EvaluatorJobConfiguration(
     /**
      * The list of package configuration providers to use.
      */
-    val packageConfigurationProviders: List<ProviderPluginConfiguration> = emptyList(),
+    val packageConfigurationProviders: List<ProviderPluginConfiguration>? = null,
```
Comment (oheger-bosch) [inline model/src/commonMain/kotlin/JobConfigurations.kt:194]: Analogously to my other comment.

File: core/build.gradle.kts [hunk 4]
```diff
@@ -81,6 +81,7 @@ dependencies {
     implementation(libs.ktorValidation)
```
Comment (oheger-bosch) [inline core/build.gradle.kts:81]: Commit message: Is this correct "Nullable properties are [...] marked as required"? Should this be "Non-nullable properties" or "marked as optional"?

File: model/src/commonMain/kotlin/JobConfigurations.kt (hunk 2 above)
Comment (mnonnenmacher) [reply model/src/commonMain/kotlin/JobConfigurations.kt:83]: I thought about that too, but I wanted to be consistent, and so far all properties have the same types and default values in both models. We could change it for lists, but then we should change it for all of them.

Comment (oheger-bosch) [reply model/src/commonMain/kotlin/JobConfigurations.kt:83]: It was the original purpose of this mapping to separate the internal model from the API. IMHO, there can well be differences if this helps us. Avoiding unnecessary null checks would fall into this category.

**Comment by Etsija** [comment]
> @Etsija Please have a look at the last commit which contains some small changes to the UI code.

LGTM!

File: model/src/commonMain/kotlin/JobConfigurations.kt (hunk 2 above)
Comment (mnonnenmacher) [reply model/src/commonMain/kotlin/JobConfigurations.kt:83]: I took a look into making more properties non-nullable in the model module, but that lead me to lot of questions how to handle some inconsistencies we already have in the job configurations. I don't want to go down this rabbit hole right now, so I'll keep the model properties non-nullable in this PR, even if it makes the current state of the mapping even more inconsistent. At least I could make some cleanups during the investigation for which I will make a separate PR after this is merged.

Comment (mnonnenmacher) [reply model/src/commonMain/kotlin/JobConfigurations.kt]: @oheger-bosch Done.

File: api/v1/model/src/commonMain/kotlin/OptionalValue.kt (hunk 1 above)
Comment (mnonnenmacher) [reply api/v1/model/src/commonMain/kotlin/OptionalValue.kt:97]: The new JSON schema generator uses reflection by default:
https://github.com/SMILEY4/schema-kenerator/tree/develop/schema-kenerator-reflection

We could try switching to the kotlinx.serialization implementation, but I don't want to do it in this PR:
https://github.com/SMILEY4/schema-kenerator/tree/develop/schema-kenerator-serialization

Comment (sschuberth) [reply api/v1/model/src/commonMain/kotlin/OptionalValue.kt:97]: > The new JSON schema generator uses reflection by default:

Ouch 🙄

**Pull request merged**